Geometry_Engine/Query/Centroid.cs

```diff
@@ -26,16 +26,72 @@
 using System;
 using BH.Engine.Reflection;
 using BH.oM.Reflection.Attributes;
+using BH.oM.Geometry.ShapeProfiles;
+using BH.oM.Base;
+using System.ComponentModel;
 
 namespace BH.Engine.Geometry
 {
     public static partial class Query
     {
+        /***************************************************/
+        /**** Public Methods - Surfaces                 ****/
+        /***************************************************/
+
+        //TODO: Remove when PlanarSurface will implement IElement2D
+        [Description("Queries the centre of area for a PlanarSurface.")]
+        [Input("surface", "The PlanarSurface to get the centre of area of.")]
+        [Input("tolerance", "Distance tolerance, default set to BH.oM.Geometry.Tolerance.Distance")]
+        [Output("centroid", "The Point at the centre of given PlanarSurface.")]
+        public static Point Centroid(this PlanarSurface surface, double tolerance = Tolerance.Distance)
+        {
+            return Centroid(new List<ICurve> { surface.ExternalBoundary }, surface.InternalBoundaries, tolerance);
+        }
+
+
+        /***************************************************/
+        /**** Public Methods - Curves lists             ****/
+        /***************************************************/
+
+        [Description("Queries the combined centre of area enclosed by a set of ICurves with an optional set of openings.\nTo give a correct result all input curves must be planar, coplanar, closed and non-self-intersecting.\nOpening curves should also be inside of outline curves.")]
+        [Input("outlines", "Set of planar, coplanar, closed and non-self-intersecting ICurves to get the combined centre of area of.")]
+        [Input("openings", "Set of planar, coplanar, closed and non-self-intersecting ICurves illustrating openings in initial set of outlines.")]
+        [Input("tolerance", "Distance tolerance, default set to BH.oM.Geometry.Tolerance.Distance")]
+        [Output("centroid", "The Point at the centre of given geometry.")]
+        public static Point Centroid(this IEnumerable<ICurve> outlines, IEnumerable<ICurve> openings = null, double tolerance = Tolerance.Distance)
+        {
+            Point centroid = new Point();
+            double area = 0;
+
+            foreach (ICurve outline in outlines.BooleanUnion(tolerance))
+            {
+                double outlineArea = outline.IArea();
+                centroid += (outline.ICentroid() * outlineArea);
+                area += outlineArea;
+            }
+
+            foreach (ICurve opening in openings.BooleanUnion(tolerance))
```

Comment:
need to null check the list of openings before this is being hit. Especially important as the method allows for default nulls as input. something like `openings  = openings ?? new List<ICurve>();` before this is run could work.